app/test_main.py

```diff
@@ -2,0 +1,55 @@
+import unittest
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+class TestAccess(unittest.TestCase):
+    @patch("app.main.has_internet_connection")
+    @patch("app.main.valid_google_url")
+    def test_valid_url_has_connection(
+            self,
+            mock_valid_google_url: bool,
+            mock_has_internet_connection: bool
+    ) -> None:
+
+        mock_valid_google_url.return_value = True
+        mock_has_internet_connection.return_value = True
+        self.assertEqual(
+            can_access_google_page("https://www.google.com"), "Accessible"
+        )
+
+    @patch("app.main.valid_google_url")
+    def test_invalid_url(
+            self,
+            mock_valid_google_url: bool
+    ) -> None:
+
+        mock_valid_google_url.return_value = False
+        self.assertEqual(
+            can_access_google_page("https://www.google.com"), "Not accessible"
+        )
+
+    @patch("app.main.has_internet_connection")
+    def test_not_internet_connection(
```

Comment:
The test name 'test_not_internet_connection' could be more descriptive. Consider renaming it to something like 'test_no_internet_connection_returns_not_accessible' to better reflect the test's purpose.